kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterServiceDependentResource.java

```diff
@@ -72,28 +73,49 @@ public Map<String, Service> desiredResources(
                 .flatMap(ProxyNetworkingModel.ClusterNetworkingModel::requiredSniLoadbalancerPorts)
                 .distinct().sorted().toList();
 
-        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts);
+        Set<BootstrapServersAnnotation.BootstrapServer> bootstraps = getLoadBalancerServiceBootstrapServers(clusterNetworkingModels);
+
+        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts, bootstraps);
 
         return Stream.concat(serviceStream, sniServiceStream).collect(toByNameMap());
     }
 
-    private ObjectMeta serviceMetadata(KafkaProxy primary, String name) {
-        return new ObjectMetaBuilder()
+    /**
+     * Get the bootstrap servers hosted by the shared LoadBalancer Service
+     */
+    private static Set<BootstrapServersAnnotation.BootstrapServer> getLoadBalancerServiceBootstrapServers(List<ProxyNetworkingModel.ClusterNetworkingModel> clusterNetworkingModels) {
+        return clusterNetworkingModels.stream()
+                .flatMap(ClusterServiceDependentResource::getBootstrapServers)
+                .collect(Collectors.toSet());
+    }
+
+    private static Stream<BootstrapServersAnnotation.BootstrapServer> getBootstrapServers(ProxyNetworkingModel.ClusterNetworkingModel networking) {
+        return networking.clusterIngressNetworkingModelResults().stream()
+                .map(ProxyNetworkingModel.ClusterIngressNetworkingModelResult::clusterIngressNetworkingModel)
+                .filter(clusterIngressModel -> clusterIngressModel.requiredSniLoadBalancerServicePorts()
+                        .findAny().isPresent())
```

Comment:
Should this be a method on `clusterIngressModel` that just returns the boolean?

Reply:
this is replaced with `Optional<SharedLoadBalancerServiceRequirements>` which we can flatMap our bootstrap servers out of.